### PR TITLE
[AIRAVATA-2509] Account confirmation, password reset emails not making it to user

### PR DESCRIPTION
**Pull request by neeraj-lad**
1. Now we use the recipient's full name (first name and last name) along with their email address to send the email. This reduces the likelihood of the mail being marked as spam.
screen shot 2017-11-16 at 17 58 27

2. Added another field 'E-mail (again)' to 'Create New Test Drive Account' page to make sure the user does not make a typo in the email address. Added code to make sure 'E-mail' and 'E-mail (again)' fields match. If there is a problem with the data validation, the user has to re-enter the 'E-mail' and 'E-mail (again)' fields along with 'Password' and 'Password (again)' fields.
screen shot 2017-11-16 at 17 57 40

### CODE REVIEW

File: app/libraries/EmailUtilities.php [hunk 1]
```diff
@@ -115,13 +133,14 @@ public static function gatewayUpdateMailToProvider($email, $gatewayId){
         $body = str_replace("\$url", URL::to('/') . '/admin/dashboard', $body);
         $body = str_replace("\$gatewayId", $gatewayId, $body);
 
+        $recipient['email'] = $email;
```
[the remark this answers is not in the document]
Comment (neeraj-lad) [reply app/libraries/EmailUtilities.php]: Yes, it works (I've used the same code on lines 87-89 for `sendPasswordResetEmail()` and I was able to successfully test it). But after reading about it a bit, it seems that it preferable to declare the array before adding key-value pairs to it, so I'll explicitly declare the array over here and the other places.

File: app/controllers/AccountController.php [hunk 2]
```diff
@@ -52,7 +53,7 @@ public function createAccountSubmit()
 
         if (Keycloak::usernameExists($username)) {
             return Redirect::to("create")
-                ->withInput(Input::except('password', 'password_confirm'))
+                ->withInput(Input::except('password', 'confirm_password', 'email', 'confirm_email'))
```
[the remark this answers is not in the document]
Comment (neeraj-lad) [reply app/controllers/AccountController.php]: So do you want me to remove only line 56 (and not line 44 - which might be caused by mismatch in emails)?

[the remark this answers is not in the document]
Comment (machristie) [reply app/controllers/AccountController.php]: Right

File: app/libraries/EmailUtilities.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (machristie) [reply app/libraries/EmailUtilities.php]: Thanks

**Comment by neeraj-lad** [comment]
@machristie I've made the changes as per our discussion.

**Comment by machristie** [comment]
@neeraj-lad can you rebase this branch against master and then update this pr (not sure if that works) or create a new pull request to merge into master?

**Comment by machristie** [comment]
Merged to develop branch. Deploying to develop environment now.

**Pull request merged**